### ISSUE TITLE
External monitor unplug crashes kitty

### ISSUE THREAD

**Issue by davc0n** [hyprwm/Hyprland#5840]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: thinkpad
Release: 6.8.8-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Sun, 28 Apr 2024 15:59:47 +0000


GPU information: 
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Picasso/Raven 2 [Radeon Vega Series / Radeon Vega Mobile Series] [1002:15d8] (rev d2) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

```

</details>

### Bug or Regression?

Bug

### Description

External monitor unplug crashes kitty if two windows are split horizontally and it has been lanched with `--single-instance` option.
If there are only two windows and a vertical split, doesn't crash.

### How to reproduce

1. Set `$terminal = kitty --single-instance` in hyprland conf
2. Open two kitty windows in an external monitor's workspace
3. Split horizontally
4. Unplug the monitor

**Comment by thejch** [comment]
could be https://github.com/hyprwm/Hyprland/issues/509
try git

**Comment by davc0n** [comment]
Solved in v0.40.0
